### ISSUE TITLE
[Feature]: Break into smaller commits

### ISSUE THREAD

**Issue by Christopher-Hayes** [di-sukharev/opencommit#114]
### Description

Right now, `oc` generates one commit with a message relevant to one change/file, and the git description will mention multiple files with different changes.

For example, here is a git description that was generated:

```
└  Commit message:
——————————————————
🐛 fix(chatgpt-view-provider.ts): move webpackScript script tag below the root div
🐛 fix(chat.tsx): add optional chaining to marked.setOptions() call
🎨 style(main.css): add text color to [role="tooltip"] element
The script tag for webpackScript was moved below the root div to ensure that the root div is loaded before the script. The optional chaining operator was added to the marked.setOptions() call to prevent errors when the marked library is not loaded. The text color was added to the [role="tooltip"] element to improve readability.
```

### Suggested Solution

One option is to break each file into its own separate commits. Not really the ideal way to use git and would need some hard-limits on how many commits it will make at once, but the commit messages would be more useful, and it seems relatively straight-forward if GPT's response is consistent.

### Alternatives

Another, more advanced option is to have GPT step through `git add --patch`. It would take some tweaking, but it would allow GPT to intelligently break up changes into separate commits if a file has changes that would logically be separated into different commits.

### Additional Context

This would make sense as an option in the CLI, so by default it would be off. Users using git properly wouldn't be running commit for multiple, different code changes. But, since this doesn't always happen in practice, this is a great opportunity for GPT ai to do the tedious work of grouping similar changes into separate commits.

**Comment by di-sukharev** [comment]
you could just stash and `oc` feature files separately, no?

**Comment by weeryan17** [comment]
You can, however like mentioned it doesn't generate a summary of all changes, so you'd have to go in and amend the commit message every time, not to mention I noticed default behavior is to push the committed code so if you accidentally select yes to that then you'd have more problems. That's why at the very least it would be nice if their was an option to output it's generated messages instead of immediately committing them.

**Comment by Christopher-Hayes** [comment]
Yeah this can all be done manually, the idea is developers are lazy and ai could potentially automate the tedious part of separating files/code into individual commits.

**Comment by kamushadenes** [comment]
I came looking exactly for that, my projects usually require lots of changes for a single feature and the commit message ends up being huge, plus things like `goreleaser` get confused when generating release notes for such large (and mixed-content) commit messages.

**Comment by Christopher-Hayes** [comment]
@kamushadenes that's a cool project. Looks further along than the other "AGI" task-based projects I've seen on GitHub.

**Comment by kamushadenes** [comment]
@Christopher-Hayes thanks! This has been my pet project for a couple weeks now. I would love to hear some feedback and ideas if you have ;)


Regarding the release notes, I ended up doing this: https://github.com/kamushadenes/chloe/blob/main/.github/scripts/release-notes.py

Not perfect, but seems to handle well this specific case.

For example, it went from this:

![image](https://user-images.githubusercontent.com/242529/230435198-704cc295-d9e7-473e-b429-0f45c2af7891.png)


To this:

![image](https://user-images.githubusercontent.com/242529/230437480-6cf0b35d-30e0-4440-938b-a78bdc61e268.png)

**Comment by dasheck0** [comment]
Is this issue resolved? TBH this is the only issue preventing me from using this regularly. I tried some of the work arounds (commit one by one, using the action, trying to resolve this in the changelog). None of them are clean enough for me to work :D So just wondering what your thoughts on this are.
